### PR TITLE
test: Add more call tests

### PR DESCRIPTION
**Pull request by chfast**

### CODE REVIEW

File: test/unittests/execute_call_test.cpp [hunk 1]
```diff
@@ -58,6 +58,34 @@ TEST(execute_call, call_with_arguments)
...
 }
 
+TEST(execute_call, call_shared_stack_space)
+{
+    /* wat2wasm
+    (module
+      (func $double (param i32) (result i64)
+        local.get 0
+        i64.extend_i32_u
+        local.get 0
```
Comment (axic) [inline test/unittests/execute_call_test.cpp:68]: Is the point of this test that the same local is used twice?

File: test/unittests/execute_call_test.cpp [hunk 2]
```diff
@@ -205,6 +233,36 @@ TEST(execute_call, call_indirect_uninited_table)
     EXPECT_THAT(execute(module, 3, {4}), Traps());
 }
 
+TEST(execute_call, call_indirect_shared_stack_space)
```
Comment (axic) [inline test/unittests/execute_call_test.cpp:236]: Seems to be a dup of `imported_functions_call_indirect` ?

Comment (chfast) [reply test/unittests/execute_call_test.cpp:236]: Both are simplified variants of it. They can detect an implementation defect in #572, so I keep them as they are.

File: test/unittests/execute_call_test.cpp [hunk 3]
```diff
@@ -58,6 +58,34 @@ TEST(execute_call, call_with_arguments)
     EXPECT_THAT(execute(parse(wasm), 1, {}), Result(4));
 }
 
+TEST(execute_call, call_shared_stack_space)
+{
+    /* wat2wasm
+    (module
+      (func $double (param i32) (result i64)
```
Comment (gumb0) [inline test/unittests/execute_call_test.cpp:65]: It's hard to guess what these tests check... Is `i64` result required here? If not, maybe make it `i32` for simplicity and more focused test?

**Review by gumb0 — APPROVED** [review]

**Pull request merged**